### PR TITLE
Add a test and fix that we don't wait for tasks, just outputs

### PR DESCRIPTION
**Pull request by Frassle**
# Description

Fixes https://github.com/pulumi/pulumi/issues/6762.

## Checklist

- [x] I have run `make tidy` to update any new dependencies
- [x] I have run `make lint` to verify my code passes the lint check
  - [x] I have formatted my code using `gofumpt`

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have run `make changelog` and committed the `changelog/pending/<file>` documenting my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Cloud API version

### CODE REVIEW

File: changelog/pending/20240326--sdk-python--the-python-sdk-only-waits-for-pending-outputs-not-all-pending-asyncio-tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changes:
+- type: fix
+  scope: sdk/python
+  description: The Python SDK only waits for pending outputs, not all pending asyncio tasks
```
Comment (tgummerer) [inline changelog/pending/20240326--sdk-python--the-python-sdk-only-waits-for-pending-outputs-not-all-pending-asyncio-tasks.yaml:4]:
```suggestion
  description: Wait only for pending outputs in the Python SDK, not all pending asyncio tasks
```

Sorry for being a pain about these :see_no_evil:

File: sdk/python/lib/pulumi/runtime/stack.py [hunk 2]
```diff
@@ -104,36 +96,23 @@ async def wait_for_rpcs(await_all_outstanding_tasks=True) -> None:
 
         # If the RPCs have successfully completed, now await all remaining outstanding tasks.
         if await_all_outstanding_tasks:
-            outstanding_tasks = _get_running_tasks()
-            if len(outstanding_tasks) == 0:
-                log.debug("No outstanding tasks to complete")
-            else:
+            while len(SETTINGS.outputs) > 0:
+                await asyncio.sleep(0)
```
Comment (tgummerer) [inline sdk/python/lib/pulumi/runtime/stack.py:100]: Out of curiosity, why is this necessary? Is this some pythonism I don't know about?

**Review by tgummerer — APPROVED** [review]

File: changelog/pending/20240326--sdk-python--the-python-sdk-only-waits-for-pending-outputs-not-all-pending-asyncio-tasks.yaml (hunk 1 above)
Comment (Frassle) [reply changelog/pending/20240326--sdk-python--the-python-sdk-only-waits-for-pending-outputs-not-all-pending-asyncio-tasks.yaml:4]: It's not a pain

File: sdk/python/lib/pulumi/runtime/stack.py (hunk 2 above)
Comment (Frassle) [reply sdk/python/lib/pulumi/runtime/stack.py:100]: It's so other things in the event loop can run. Just read it like "thread yield"
